test/test.js

```diff
@@ -397,6 +397,26 @@ QUnit.test( "submitHandler keeps submitting button", function( assert ) {
 	$( "#userForm" ).submit();
 } );
 
+QUnit.test( "submitHandler keeps submitting button, even if descendants are clicked", function( assert ) {
+	var button;
+
+	$( "#testForm27" ).validate( {
+		debug: true,
+		submitHandler: function( form ) {
+
+			// Dunno how to test this better; this tests the implementation that uses a hidden input
+			var hidden = $( form ).find( "input:hidden" )[ 0 ];
+			assert.deepEqual( hidden.value, button.value );
+			assert.deepEqual( hidden.name, button.name );
+
+			return false;
+		}
+	} );
+	$( "#testForm27 [name=\"year\"]" ).val( "2016" );
+	button = $( "#testForm27 :submit" )[ 0 ];
+	$( button ).find( "span" ).click();
+} );
+
```

Comment:
You can extend your tests by testing the property `submitButton` of the validator. So, your tests will look like this:
```js
var button = $( "#testForm27 :submit" )[ 0 ];
var v = $( "#testForm27" ).validate( {
	debug: true,
	submitHandler: function( form ) {

		// Compare the button with the `submitButton` property
		assert.deepEqual(
			v.submitButton, button, "The submitButton property should be the same as button"
		);

		var hidden = $( form ).find( "input:hidden" )[ 0 ];
		assert.deepEqual( hidden.value, button.value );
		assert.deepEqual( hidden.name, button.name );

		return false;
	}
} );

$( "#testForm27 [name=\"year\"]" ).val( "2016" );

$( button ).find( "span" ).click();
```